### PR TITLE
Enable circleci build.

### PR DESCRIPTION
**Pull request by nomaddo**
This pullreq enable circleci auto-build.

**NOTICE**
This pullreq require extra configuration in addition to ordinary setting up to `circleci`: please define

- CIRCLECI_API_TOKEN

after setting up and get the token of your `VC4C` page of circleci.
See the API doc: https://circleci.com/docs/api/v1-reference/#summary

This is for downloading latest artifacts of `VC4C`.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    docker:
+      - image: nomaddo/cross-rpi
+    steps:
+      - checkout
+      - run:
+          name: download header
+          command: git clone https://github.com/doe300/VC4CLStdLib.git VC4CLStdLib
+      - run:
+          name: get libVC4CC.so url
+          command: curl "https://circleci.com/api/v1.1/project/github/doe300/VC4C/latest/artifacts?circle-token=${CIRCLECI_API_TOKEN}=master&filter=successful" --output /tmp/dump
```
Comment (doe300) [inline .circleci/config.yml]: On my local machine, I can download this file (and the artifact files) without any CircleCI API-key. Do you know if there is anything wrong with my CircleCI configuration or does it generally work without an API token?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: nomaddo/cross-rpi
+    steps:
+      - checkout
+      - run:
+          name: download header
+          command: git clone https://github.com/doe300/VC4CLStdLib.git VC4CLStdLib
```
Comment (doe300) [inline .circleci/config.yml]: You should not need the VC4CLStdLib to compile VC4CL

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      - run:
+          name: configure
+          command: cmake . -DCROSS_COMPILE=ON -DBUILD_TESTING=ON -DCMAKE_FIND_DEBUG_MODE=1  -DOpenCL_INCLUDE_DIR=/opt/rasberrypi/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/arm-linux-gnueabihf/include -DOpenCL_LIBRARY=/opt/rasberrypi/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/arm-linux-gnueabihf/include -DCMAKE_INSTALL_PREFIX=/usr -DVC4CL_STDLIB_HEADER_SOURCE=./VC4CLStdLib/include/VC4CLStdLib.h -DVC4CC_LIBRARY=libVC4CC.so
+      - run:
+          name: build
+          command: make -j2
+      - run:
+          name: deb-packing
+          command: cpack -G DEB
+      - store_artifacts:
```
Comment (doe300) [inline .circleci/config.yml]: The created artifacts are wrong, these are the artifacts from VC4C, not VC4CL. VC4CL has following artifacts: `libVC4CL.so.0.4`, `tools/v3d_info` and `tools/v3d_profile`

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+          path: build/libVC4CC.so.0.4
+          distination: libVC4CC.so.0.4
+      - store_artifacts:
+          path: build/VC4C
+          path: VC4C
+      - store_artifacts:
+          path: build/test/TestVC4C
+          distination: TestVC4C
+      - store_artifacts:
```
Comment (doe300) [inline .circleci/config.yml]: Same here, the wrong debian package is stored

File: CMakeLists.txt [hunk 5]
```diff
@@ -15,6 +15,8 @@ option(REGISTER_POKE_KERNELS "Use the faster way of register to run kernels" ON)
 option(BUILD_DEBUG "Build with debugging symbols. Otherwise build for performance" OFF)
 # Option to enable/disable cross compilation
 option(CROSS_COMPILE "Cross compile for Raspbian" OFF)
+# Option whether to create deb package
```
Comment (doe300) [inline CMakeLists.txt]: Can you re-base your PR, so this file is not changed twice? If that is too much trouble, I could pull this version, github seems to be able to resolve duplicate identical changes.

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+          command: git clone https://github.com/doe300/VC4CLStdLib.git VC4CLStdLib
+      - run:
+          name: get libVC4CC.so url
+          command: curl "https://circleci.com/api/v1.1/project/github/doe300/VC4C/latest/artifacts?circle-token=${CIRCLECI_API_TOKEN}=master&filter=successful" --output /tmp/dump
+      - run:
+          name: get libVC4CC.so
+          command: wget -O libVC4CC.so $(python tools/get_url.py "/tmp/dump")
+      - run:
+          name: configure
+          command: cmake . -DCROSS_COMPILE=ON -DBUILD_TESTING=ON -DCMAKE_FIND_DEBUG_MODE=1  -DOpenCL_INCLUDE_DIR=/opt/rasberrypi/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/arm-linux-gnueabihf/include -DOpenCL_LIBRARY=/opt/rasberrypi/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/arm-linux-gnueabihf/include -DCMAKE_INSTALL_PREFIX=/usr -DVC4CL_STDLIB_HEADER_SOURCE=./VC4CLStdLib/include/VC4CLStdLib.h -DVC4CC_LIBRARY=libVC4CC.so
```
Comment (doe300) [inline .circleci/config.yml]: We install VC4CLStdLib and VC4CL into `/usr/local/`, shouldn't we then also install VC4CL into `/usr/local`?

Comment (doe300) [reply .circleci/config.yml]: Another thing: For VC4CL to find VC4C, the library and the headers need to be found. Wouldn't it be easier to download vc4c-xxx.deb, install it and use its headers and library?
I changed CMakeLists.txt, so it should automatically find the library and headers in the installed directories.

File: .circleci/config.yml (hunk 1 above)
Comment (nomaddo) [reply .circleci/config.yml]: Mmm... same as my local machie. 
I am not sure why, but currently remove them.

**Comment by doe300** [comment]
As a note: I just completed the postinst and prerem scripts for installing VC4C (see [here](https://github.com/doe300/VC4C/issues/9)), so it would be possible to simply download the `vc4cl-stdlib` and `vc4c` debian packages and installing them. This should provide the correct paths to find the compiler in.

**Pull request merged**